pkg/exporter/syslogexporter/factory.go

```diff
@@ -0,0 +1,64 @@
+// Copyright 2023 OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package syslogexporter
+
+import (
+	"context"
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/exporter"
+	"go.opentelemetry.io/collector/exporter/exporterhelper"
+)
+
+const (
+	// The value of "type" key in configuration.
+	typeStr        = "syslog"
+	stabilityLevel = component.StabilityLevelAlpha
+)
+
+// NewFactory returns a new factory for the syslog exporter.
+func NewFactory() exporter.Factory {
+	return exporter.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		exporter.WithLogs(createLogsExporter, stabilityLevel),
+	)
+}
+
+func createDefaultConfig() component.Config {
+	qs := exporterhelper.NewDefaultQueueSettings()
+	qs.Enabled = false
+
+	return &Config{
+		Endpoint: DefaultEndpoint,
+		Port:     DefaultPort,
+		Format:   DefaultFormat,
+		Protocol: DefaultProtocol,
+	}
+}
+
+func createLogsExporter(
+	ctx context.Context,
+	params exporter.CreateSettings,
+	cfg component.Config,
+) (exporter.Logs, error) {
+	exp, err := newLogsExporter(ctx, params, cfg.(*Config))
+	if err != nil {
+		return nil, fmt.Errorf("failed to create the logs exporter: %w", err)
+	}
+
+	return exp, nil
+}
```

Comment:
Is there any reason to not use exporterhelper like for example in fileexporter?
https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/e506d194e43c07e8deb326d475a97173c471dc8f/exporter/fileexporter/factory.go#L77-L83

Reply:
nvm, newLogsExporter does it